### DISCUSSION TITLE
Is there any plan to support multi-tenancy in duckdb?

### DISCUSSION THREAD

**Discussion by zhangxinyu1** [duckdb/duckdb#3717] [General]
For better resource isolation, should we support multi-tenancy?

**Comment by Alex-Monahan** [comment]
Hello!

A common pattern I've seen in SQLite is to have a separate DB file per tenant. That would work well in DuckDB also! The only missing piece is either a DuckDB reader or support for the Attach keyword.  Prior discussion of that is here: https://github.com/duckdb/duckdb/issues/1985
